### PR TITLE
fix(packages/graph-client): use promise allSettled instead of promise…

### PR DESCRIPTION
**Pull request by chillichelli**
… all

### CODE REVIEW

File: packages/graph-client/.graphclientrc.yml [hunk 1]
```diff
@@ -132,26 +132,26 @@ additionalTypeDefs: |
...
       chainIds: [Int!]!,
-    ): [ChefUser!]!
+    ): [ChefUser]
     crossChainUserWithFarms(
       id: ID!,
       block: Block_height,
       now: Int!,
       chainIds: [Int!]!,
-    ): [UserWithFarm!]
+    ): [UserWithFarm]
```
[the remark this answers is not in the document]
Comment (chillichelli) [reply packages/graph-client/.graphclientrc.yml]: You cant use Promise.allSettled without allowing an empty response

File: packages/graph-client/resolvers/sushiswap/crossChainChefUser.ts [hunk 2]
```diff
@@ -45,5 +45,13 @@ export const crossChainChefUser: QueryResolvers['crossChainChefUser'] = async (r
       .map((chainId) =>
         fetcher({ chainId, subgraphName: MINICHEF_SUBGRAPH_NAME[chainId], subgraphHost: SUBGRAPH_HOST[chainId] })
       ),
-  ]).then((users) => users.flat())
+  ]).then((users) => {
```
[the remark this answers is not in the document]
Comment (matthewlilley) [reply packages/graph-client/resolvers/sushiswap/crossChainChefUser.ts:48]: Even worse is mixing the two, stick to then until we decide to refactor it all.

[the remark this answers is not in the document]
Comment (matthewlilley) [reply packages/graph-client/resolvers/sushiswap/crossChainChefUser.ts:48]: also awaiting within promise chains can block in some cases where you might not want it to, so there's that too, you want promises to be executed in parellel a lot of the time

File: packages/graph-client/.graphclientrc.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (LufyCZ) [reply packages/graph-client/.graphclientrc.yml]: you filter the empty responses out (and if all all empty return an empty array), there's no reason to burden the consumers with null checks

**Pull request merged**